semeio/workflows/localisation/localisation_config.py

```diff
@@ -149,6 +233,74 @@ class ExponentialConfig(GaussianConfig):
     method: Literal["exponential_decay"]
 
 
+class ScalingFromFileConfig(BaseModel):
+    method: Literal["from_file"]
+    filename: str
+    param_name: str
+
+
+class ScalingForSegmentsConfig(BaseModel):
+    method: Literal["segment"]
+    segment_filename: str
+    param_name: str
+    active_segments: conlist(item_type=conint(ge=0))
+    scalingfactors: conlist(item_type=confloat(ge=0.0))
+    smooth_ranges: Optional[conlist(item_type=conint(ge=0), min_items=2, max_items=2)]
+
+    @validator("active_segments", pre=True)
+    def validate_active_segments(cls, values):
+
+        if isinstance(values, int):
+            active_segments = [values]
+        elif isinstance(values, list):
+            active_segments = values
+            for v in active_segments:
+                if v < 0:
+                    raise ValueError(
+                        "Active segments must be integers >=0"
+                        " in keyword 'active_segments.'"
+                    )
+        return active_segments
+
+    @validator("scalingfactors", pre=True)
+    def validate_scalingfactors(cls, values):
+        if isinstance(values, float):
+            if not 0 <= values <= 1:
+                raise ValueError(
+                    "Scaling factors must be in interval [0.0 ,1.0] "
+                    "in keyword 'segment.'"
+                )
+            scalingfactors = [values]
+        elif isinstance(values, list):
+            scalingfactors = values
+            for v in scalingfactors:
+                if not 0 <= v <= 1:
+                    raise ValueError(
+                        "Scaling factors must be in interval [0.0 ,1.0] "
+                        "in keyword 'segment.'"
+                    )
+        else:
+            scalingfactors = None
+        return scalingfactors
+
+    @validator("smooth_ranges", pre=True)
+    def validate_smooth_ranges(cls, values):
```

Comment:
Not needed, use:
```
    smooth_ranges: Optional[conlist(item_type=conint(ge=0), min_items=2, max_items=2)]
```

Reply:
But if you would just like to change the error message you can do that afterwards, something like this: https://github.com/samuelcolvin/pydantic/issues/322, that would be better that to have duplicate validation which will very easily come out of sync.

Reply:
For me this looks a bit complicated. I suggest to avoid double validation that may get out of sync to keep only one of them so that when I want more detailed error message, I use my own validation function and when I am satisfied with the autogenerated error message, I don't check for the same thing in the validation function or drop using the validation function for that field. So this means that I may remove conlist(...) or similar syntax some places and keep it other places where it is used.

Reply:
Have now updated the validation to avoid double validation.